src/ic-cdk/src/api/call.rs

```diff
@@ -476,16 +477,26 @@ pub fn msg_cycles_accept128(max_amount: u128) -> u128 {
 }
 
 /// Returns the argument data as bytes.
-pub(crate) unsafe fn arg_data_raw() -> Vec<u8> {
-    let len: usize = ic0::msg_arg_data_size() as usize;
-    let mut bytes = vec![0u8; len as usize];
-    ic0::msg_arg_data_copy(bytes.as_mut_ptr() as i32, 0, len as i32);
-    bytes
+pub fn arg_data_raw() -> Vec<u8> {
+    unsafe {
+        let len: usize = ic0::msg_arg_data_size() as usize;
+        let mut bytes = vec![0u8; len as usize];
+        ic0::msg_arg_data_copy(bytes.as_mut_ptr() as i32, 0, len as i32);
+        bytes
```

Comment:
I usually prefer the following pattern:
```suggestion
        let mut bytes = Vec::with_capacity(len);
        ic0::msg_arg_data_copy(bytes.as_mut_ptr() as i32, 0, len as i32);
        bytes.set_len(len);
        bytes
```
The reasoning is that we don't need to initialize the vector with zeros since we're going to overwrite the buffer anyway. I never measured the difference in cycle consumption, though.
I used this approach here, for example: https://sourcegraph.com/github.com/dfinity/ic@afa6f02ac60f0c1ab0eae76d188a7ce97558e2dc/-/blob/rs/rust_canisters/dfn_core/src/stable.rs?L88-95

Reply:
Makes sense, thanks!